roles/postgres/tasks/restore.yml

```diff
@@ -0,0 +1,33 @@
+- name: Restore database backups if necessary
+  become: true
+  become_user: postgres
+  block:
+    - name: Check if we need to restore a database backup
+      community.postgresql.postgresql_query:
+        db: "{{ item.database }}"
+        query: "SELECT * FROM pg_catalog.pg_tables WHERE tableowner = %s"
+        positional_args:
+          - "{{ item.username }}"
+      register: existing_tables
+
+    - name: Copy and restore a database backup
+      when: existing_tables.rowcount == 0
```

Comment:
Is this to prevent restoration when the database is not empty?

Reply:
Correct! Let me add a comment.